lib/janus/connection.js

```
@@ -107,15 +107,18 @@ JanusConnection.prototype.onTimeout = function(message) {
 };
 
 JanusConnection.prototype.onRemove = function() {
-  this._removeSession();
-  serviceLocator.get('logger').info('Removed ' + this);
+  return this._removeSession().then(function() {
+    serviceLocator.get('logger').info('Removed ' + this);
+  }.bind(this));
 };
 
 JanusConnection.prototype._removeSession = function() {
   if (null !== this.session) {
-    this.session.onRemove();
-    this.session = null;
+    return this.session.onRemove().finally(function() {
```

Still finally. We don't want to catch/ignore exception here as well.